### DISCUSSION TITLE
Authelia Support / Nginx Modules Customization

### DISCUSSION THREAD

**Discussion by EthanShoeDev** [caprover/caprover#1562] [Q&A] [answered]
Hi,
I was wondering if anyone has been able to get [Authelia ](https://www.authelia.com/)to work with the CapRover?

I believe I should be able to set everything up using the documentation here: https://www.authelia.com/integration/proxies/nginx/
However, one line makes things a bit more complicated:

In the [requirements section](https://www.authelia.com/integration/proxies/nginx/#requirements), it states:

- [NGINX](https://www.nginx.com/) must be built with the http_set_misc module or the nginx-mod-http-set-misc package if you want to preserve more than one query parameter when redirected to the portal due to a limitation in [NGINX](https://www.nginx.com/)
     
     
First I tried to investigate if the module is included in my running captain deployed nginx container. Consoled into the container and ran:

    $ nginx -V 2>&1 | tr ' ' '\n'

The output revealed that the module is not included.

Running the same command in the linuxserver.io nginx container revealed: 

    --add-dynamic-module=/home/buildozer/aports/main/nginx/src/set-misc-nginx-module-0.32/

Is there any way that I can add the module myself or would that have to be done within the cap rover code itself?

**Comment by githubsaturn** [comment, marked as answer]
Most config constants (including the nginx image) can be override: https://caprover.com/docs/troubleshooting.html#customize-config-settings

Just run:
```

All you need to do is

echo  "{\"nginxImageName\":\"mycustom/nginx-image:latest\"}" >  /captain/data/config-override.json
docker service update captain-captain --force
```

Of course, change `mycustom/nginx-image:latest` to the actual custom image that you want.

**Comment by hAhl2** [comment]
Hello @SherlockShoe , were you able to get an 'Authelia' instance up and running on caprover?  I'd eventually would like to set one up to.
